### PR TITLE
Menu component

### LINKED ISSUE #89: Menu component
# UI

Screen Shot 2562-11-14 at 10 44 05

# Usage

Describe how to use the component

```html
<!-- Menu with banner and menu links -->
<Menu banner="My banner" categories="[ "x1", "x2", ... ]" items="[ { key: "x1", text: "x" , to: "/"},{key: "x2", text: "xx" , to: "/"}, ... ]" />

<!-- Menu with only banner -->
<Menu banner="a" />

...
```

# Properties

List all the properties needed for the component:
- **banner**: `String` (required) -> display banner on Menu
- **categories**: `Array of string` (optional) -> Display items on each category values
- **items**: `Array of object` (optional) -> items to display on the footer

# Slots

- define each slot name using `items_key` exists on `categories`

# Events

none

### PR DESCRIPTION
**Pull request by blackinno**
- revers menu component. Closes https://github.com/mesg-foundation/mesg-components/issues/89

### CODE REVIEW

**Comment by blackinno** [comment]
https://github.com/mesg-foundation/mesg-components/issues/89

File: packages/Menu/Menu.vue [hunk 1]
```diff
@@ -0,0 +1,154 @@
...
+    </nav>
+  </div>
+</template>
+
+<script>
+export default {
+  name: 'Menu',
+  props: {
+    banner: { type: String, required: true },
+    categories: { type: Array },
```
Comment (antho1404) [inline packages/Menu/Menu.vue]: what's the categories? We already have all the items so we could create the menu with only the items we don't necessary need slots here

File: packages/Menu/Menu.vue [hunk 2]
```diff
@@ -0,0 +1,154 @@
...
+      </div>
+    </nav>
+  </div>
+</template>
+
+<script>
+export default {
+  name: 'Menu',
+  props: {
+    banner: { type: String, required: true },
```
Comment (antho1404) [inline packages/Menu/Menu.vue]: banner is not really clear, maybe logo would be better

File: packages/Menu/Menu.vue [hunk 3]
```diff
@@ -0,0 +1,154 @@
+<template>
+  <div>
+    <nav flex space-between align-center>
+      <div flex row space-between align-center>
+        <div>
+          <a>
```
Comment (antho1404) [inline packages/Menu/Menu.vue]:
```suggestion
          <a to="/">
```

**Review by antho1404 — CHANGES_REQUESTED** [review]

File: packages/Menu/Menu.vue (hunk 1 above)
Comment (blackinno) [reply packages/Menu/Menu.vue]: Fixed

File: packages/Menu/Menu.vue (hunk 2 above)
Comment (blackinno) [reply packages/Menu/Menu.vue]: rename to `logo`

File: packages/Menu/README.md [hunk 4]
```diff
@@ -0,0 +1,54 @@
+# @MESG-COMPONENTS/MENU
+
+## Install
+
+`npm install @mesg-components/menu` or `yarn add @mesg-components/menu`
+
+## Properties
+
+- **banner**: `String`, **Required**, -> _banner to display on the footer_
```
Comment (antho1404) [inline packages/Menu/README.md]: please update the documentation

File: packages/Menu/README.md [hunk 5]
```diff
@@ -0,0 +1,54 @@
...
+- **items**: `Array of object`, **Optional**, -> _items to display on the footer_
```
Comment (antho1404) [inline packages/Menu/README.md]: could you describe what kind of objects are needed here

File: packages/Menu/README.md [hunk 6]
```diff
@@ -0,0 +1,54 @@
...
+
+## Example
+
+```js
+import logo from '../assets/img/MESG-logo-horizontal-purple.svg'
+
+const categories = ['technology', 'developers', 'showcase', 'contributions', 'token', 'get-started']
+const items = [
+  {
+    key: 'technology',
```
Comment (antho1404) [inline packages/Menu/README.md]: shouldn't be necessary anymore, same for the categories array

File: packages/Menu/README.md (hunk 4 above)
Comment (blackinno) [reply packages/Menu/README.md]: Updated.

File: packages/Menu/README.md (hunk 5 above)
Comment (blackinno) [reply packages/Menu/README.md]: done.

File: packages/Menu/README.md (hunk 6 above)
Comment (blackinno) [reply packages/Menu/README.md]: done.

**Review by antho1404 — COMMENTED** [review]
This is good but I just realized something, we use nuxt most of the time and the routing is a bit different with nuxt, we need to use `<nuxt-link>` instead of `<a>`.
The link for the logo doesn't work because `to` only works with `nuxt-link` not with `a`.

2 solutions:
- We force all navigation with nuxt and replace all the links with a nuxt-link so this component can only be used in a nuxt environment (which is fine for now because we use only that)
- We add a boolean property `nuxt` to say we are in a nuxt environment and in staht case we use `nuxt-link`, otherwise we use `a`

**Comment by blackinno** [comment]
> This is good but I just realized something, we use nuxt most of the time and the routing is a bit different with nuxt, we need to use `<nuxt-link>` instead of `<a>`.
> The link for the logo doesn't work because `to` only works with `nuxt-link` not with `a`.
> 
> 2 solutions:
> 
> * We force all navigation with nuxt and replace all the links with a nuxt-link so this component can only be used in a nuxt environment (which is fine for now because we use only that)
> * We add a boolean property `nuxt` to say we are in a nuxt environment and in staht case we use `nuxt-link`, otherwise we use `a`

I used the second solution. I add the variable `isNuxt` for use NuxtJS framework.

**Comment by antho1404** [comment]
there is an issue with the nuxt-link 
Screen Shot 2019-11-19 at 19 11 01
Maybe this is normal because there is no `nuxt-link` components. Could you check to include it in a nuxt project, example mesg.com and confirm that it's working correctly

**Comment by blackinno** [comment]
> there is an issue with the nuxt-link
> Screen Shot 2019-11-19 at 19 11 01
> Maybe this is normal because there is no `nuxt-link` components. Could you check to include it in a nuxt project, example mesg.com and confirm that it's working correctly

It's normal. I created a demo nuxt project to test the components. The menu is working correctly.

Screen Shot 2562-11-20 at 09 03 21

**Review by antho1404 — APPROVED** [review]

**Comment by antho1404** [comment]
looks good to me, waiting for @EmmanuelDrouin's review

**Comment by EmmanuelDrouin** [comment]
Good for me, let's merge it

**Pull request merged**